src/db/write.rs

```diff
@@ -193,8 +193,18 @@ impl<'a> Write<'a> {
         key_prefix: &[u8],
         json_pointer: &str,
     ) -> Result<(), CreateIndexError> {
-        // TODO: This should be a nop if the specified index already exists.
         use CreateIndexError::*;
+
+        // Check to see if the index already exists.
+        if let Some(index) = self.indexes.get(name) {
+            if index.meta.definition.key_prefix == key_prefix
+                && index.meta.definition.json_pointer == json_pointer
+            {
+                return Ok(());
+            }
+            return Err(IndexExistsWithDifferentDefinition);
```

Comment:
Yeah I'm not sure what correct behavior is but this seems like a conservative starting point. Good idea.